### PR TITLE
[BSE-4882] Refactor Series method tests

### PR DESCRIPTION
**Pull request by chrisoh7**
## Changes included in this PR

Refactors test codes for Series.str, Series.dt, and Series methods and addresses duplicated code issue. 
Updates docs to note the different behaviors of Bodo and Pandas in the output values of datetime predicate methods on NaT series. 

## Testing strategy

PR CI, adds more test cases such as null series. 

## User facing changes

Documentation properly notes behavior of Bodo that differs from Pandas. 

## Checklist
- [x] Pipelines passed before requesting review. To run CI you must include `[run CI]` in your commit message.
- [x] I am familiar with the [Contributing Guide](https://github.com/bodo-ai/Bodo/blob/main/CONTRIBUTING.md) 
- [x] I have installed + ran pre-commit hooks.

### CODE REVIEW

File: bodo/tests/test_df_lib/test_series_generator.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                _test_equal(out_bodo, out_pd, check_pandas_types=False)
+            except AssertionError as e:
+                """
+                Exception case handler: currently month_name and day_name with locale arg
+                returns outputs that trivially differ from Pandas. Also, partition and rpartition returns same outputs, but 
+                with different index types causing the equality test to fail. 
+                In this case, we print out the outputs Bodo vs. Pandas for manual inspection. 
+                """
+                if name in [
+                    "partition",
```
Comment (scott-routledge2) [inline bodo/tests/test_df_lib/test_series_generator.py]: For partition/rpartition I think the issue is that we are not propagating the index arrays. We should look into fixing that. (you can look at what we do in `_get_series_python_func_plan` for example of selecting index arrays).

Comment (scott-routledge2) [reply bodo/tests/test_df_lib/test_series_generator.py]: Also instead of this manual checking, I think it would be better if we just constructed the "expected" result for each case where the result does not match pandas. For example, you could have an "expected_result" field in your input:
``` py
"partition": [
(args, kwargs, expected_results), ...
]
```

**Review by scott-routledge2 — APPROVED** [review]
LGTM, one comment looks like I meant to make on a previous PR.
Thanks @chrisoh7

File: bodo/tests/test_df_lib/test_series_generator.py (hunk 1 above)
Comment (chrisoh7) [reply bodo/tests/test_df_lib/test_series_generator.py]: Done!

**Review by ehsantn — APPROVED** [review]
Looks good to me. Thanks @chrisoh7 .
